src/items.rs

```diff
@@ -780,10 +786,23 @@ impl<'a> FmtVisitor<'a> {
     // TODO we farm this out, but this could spill over the column limit, so we
     // ought to handle it properly.
     fn rewrite_fn_input(&self, arg: &ast::Arg) -> String {
-        format!("{}: {}",
-                pprust::pat_to_string(&arg.pat),
-                pprust::ty_to_string(&arg.ty))
+        if is_named(arg) {
+            format!("{}: {}",
+                    pprust::pat_to_string(&arg.pat),
+                    pprust::ty_to_string(&arg.ty))
+        } else {
+            pprust::ty_to_string(&arg.ty)
+        }
+    }
+}
+
+fn is_named(arg: &ast::Arg) -> bool {
+    if let ast::Pat_::PatIdent(_, ident, _) = arg.pat.node {
+        if ident.node == token::special_idents::invalid {
```

Comment:
Can replace the whole if stmt with `ident.node != token::special_idents::invalid`